tests/samplers/test_tfdata_sampler.py

```diff
@@ -0,0 +1,180 @@
+
+        for i, x in enumerate(augmented_ds):
+            if i < warmup:
+                self.assertListEqual(x.numpy().tolist(), [0, 1])
+            else:
+                self.assertListEqual(x.numpy().tolist(), [12, 13])
+                break
+
+
+class TestTFDataSampler(tf.test.TestCase):
+    def setUp(self):
+        self.ds = tf.data.Dataset.from_tensor_slices(
+            (
+                tf.random.uniform((6, 2), dtype=tf.float32),
+                tf.constant([1, 1, 1, 2, 2, 2], dtype=tf.int64),
+            )
+        )
+        self.expected_elementspec = (
+            tf.TensorSpec(shape=(None, 2), dtype=tf.float32, name=None),
+            tf.TensorSpec(shape=(None,), dtype=tf.int64, name=None),
+        )
+
+    def test_cardinality_is_finite(self):
+        # Test that an exception is raised when the input dataset is infinite
+        ds = tf.data.Dataset.from_tensors([1]).repeat()
+        with self.assertRaisesWithLiteralMatch(ValueError, "Dataset must be finite."):
+            tfds.TFDataSampler(ds)
+
+    def test_cardinality_is_known(self):
+        # Test that an exception is raised when the input dataset has unknown cardinality
+        ds = tf.data.Dataset.from_tensor_slices([1, 2, 3]).shuffle(3).filter(lambda x: x > 1)
+        with self.assertRaisesWithLiteralMatch(ValueError, "Dataset cardinality must be known."):
+            tfds.TFDataSampler(ds)
+
+    def test_output_batch_size(self):
+        # Test that the output batch size is correct
+        out_ds = tfds.TFDataSampler(self.ds)
+        self.assertEqual(out_ds.element_spec, self.expected_elementspec)
+
+    def test_output_classes_per_batch(self):
+        # Test that the number of classes per batch is correct
+        out_ds = tfds.TFDataSampler(self.ds, classes_per_batch=1)
+        self.assertEqual(out_ds.element_spec, self.expected_elementspec)
+
+    def test_output_examples_per_class_per_batch(self):
+        # Test that the number of examples per class per batch is correct
+        out_ds = tfds.TFDataSampler(self.ds, examples_per_class_per_batch=1)
+        self.assertEqual(out_ds.element_spec, self.expected_elementspec)
+
+    def test_output_class_list(self):
+        # Test that the class list is correctly used
+        out_ds = tfds.TFDataSampler(self.ds, class_list=[1])
+        self.assertEqual(out_ds.element_spec, self.expected_elementspec)
+
+    def test_output_total_examples_per_class(self):
+        # Test that the total number of examples per class is correctly used
+        out_ds = tfds.TFDataSampler(self.ds, total_examples_per_class=2)
+        self.assertEqual(out_ds.element_spec, self.expected_elementspec)
+
+    def test_output_augmenter(self):
+        # Test that the augmenter is correctly applied
+        out_ds = tfds.TFDataSampler(self.ds, augmenter=lambda x, y: (x * 2, y))
+        self.assertEqual(out_ds.element_spec, self.expected_elementspec)
```

Comment:
these checks are only checking the output spec, not the actual output (size or values)
I think for some of the tests the output needs to be checked, e.g., the augmenter